apps/condo/domains/banking/hooks/useCategoryModal.tsx

```diff
@@ -0,0 +1,167 @@
+    }
+}
+
+export const useCategoryModal: IUseCategoryModal = ({
+    bankTransactions = [], bankContractorAccounts = [], type, updateSelected,
+}) => {
+    const intl = useIntl()
+    const TransactionTitle = intl.formatMessage({ id: 'global.transaction' }, { isSingular: bankTransactions.length === 1 })
+    const IncomeTitle = intl.formatMessage({ id: 'global.income' }, { isSingular: false })
+    const WithdrawalTitle = intl.formatMessage({ id: 'global.withdrawal' }, { isSingular: false })
+    const ContractorTitle = intl.formatMessage({ id: 'global.contractor' }, { isSingular: true })
+    const BankAccountTitle = intl.formatMessage({ id: 'global.bankAccount' })
+    const ChooseCategoryTitle = intl.formatMessage({ id: 'pages.banking.chooseCategory' })
+    const ContractorsSelectedTitle = intl.formatMessage({ id: 'pages.banking.categoryModal.contractorsSelected' })
+    const TransactionsSelectedTitle = intl.formatMessage({ id: 'pages.banking.categoryModal.transactionsSelected' })
+    const PaymentPurposeTitle = intl.formatMessage({ id: 'global.paymentPurpose' })
+    const SumTitle = intl.formatMessage({ id: 'global.sum' })
+    const SaveTitle = intl.formatMessage({ id: 'Save' })
+
+    const { loading, bankCostItemGroups } = useBankCostItemContext()
+
+    const [open, setOpen] = useState(false)
+    const [selectedCostItem, setSelectedCostItem] = useState(null)
+
+    const closeModal = useCallback(() => {
+        setOpen(false)
+    }, [])
+    const onGroupChange = useCallback((event) => {
+        setSelectedCostItem(event.target.value)
+    }, [])
+
+    const handleSave = useCallback(async () => {
+        if (type === 'contractor') {
+            await updateSelected({
+                variables: {
+                    data: bankContractorAccounts.map(contractor => {
+                        return {
+                            id: contractor.id,
+                            data: {
+                                costItem: { connect: { id: selectedCostItem } },
+                            },
+                        }
+                    }),
+                },
+            })
+        } else if (type === 'income' || type === 'withdrawal') {
+            await updateSelected({
+                variables: {
+                    data: bankTransactions.map(transaction => {
+                        return {
+                            id: transaction.id,
+                            data: {
+                                costItem: { connect: { id: selectedCostItem } },
+                            },
+                        }
+                    }),
+                },
+            })
+        }
+
+        setOpen(false)
+        setSelectedCostItem(null)
+    }, [selectedCostItem, type, bankContractorAccounts, bankTransactions, updateSelected])
+
+
+    const categoryModal = useMemo(() => {
+        let modalTitle
+        const listDataSource = []
+
+        if (type === 'income' || type === 'withdrawal') {
```

Comment:
For `type` variable a set of values ['income', 'withdrawal'] — is a trait, but 'contractor' is an entity.
Maybe to use here a "transaction" value instead of 'income' and 'withdrawal' to finally have a fork on entity type?